build.js

```diff
@@ -52,51 +43,45 @@ StyleDictionary.registerTransform({
   },
 });
 
+const customColorObjectFormatter = (dictionary, isJS) => {
+  const valueOrType = (token) =>
+    isJS ? `"${token.value}"` : `${getTypeScriptType(token.value)}`;
+  const declaration = isJS ? "" : `export const `;
+  const commaOrColon = isJS ? `,` : `;`;
+
+  return Object.entries(dictionary.properties.colors)
+    .map((tokens) => {
+      const colorObj = tokens[0];
+      const filteredTokens = dictionary.allTokens.filter(
+        (token) => token.attributes.type === colorObj
+      );
+
+      return (
+        declaration +
+        `${colorObj} : {` +
+        filteredTokens.map((token) => {
+          return `${token.name} : ` + valueOrType(token);
+        }) +
+        `}${commaOrColon}`
+      );
+    })
+    .join(`\n`);
+};
```

Comment:
my quick attempt of using a helper function(s) while trying to keep everything together - def open to suggestions if we can make this cleaner